classes/Product.php

```diff
@@ -1578,16 +1578,32 @@ public function updateAttribute($id_product_attribute, $wholesale_price, $price,
         $combination = new Combination($id_product_attribute);
 
         if (!$update_all_fields) {
-            $combination->setFieldsToUpdate(array(
-                'price' => !is_null($price),
-                'wholesale_price' => !is_null($wholesale_price),
-                'ecotax' => !is_null($ecotax),
-                'weight' => !is_null($weight),
-                'unit_price_impact' => !is_null($unit),
-                'default_on' => !is_null($default),
-                'minimal_quantity' => !is_null($minimal_quantity),
-                'available_date' => !is_null($available_date),
-            ));
+            $fieldsToUpdate = array();
+            if (!is_null($price)) {
+                $fieldsToUpdate['price'] = $price;
+            }
+            if (!is_null($wholesale_price)) {
+                $fieldsToUpdate['wholesale_price'] = $wholesale_price;
+            }
+            if (!is_null($ecotax)) {
+                $fieldsToUpdate['ecotax'] = $ecotax;
+            }
+            if (!is_null($weight)) {
+                $fieldsToUpdate['weight'] = $weight;
+            }
+            if (!is_null($unit)) {
+                $fieldsToUpdate['unit_price_impact'] = $unit;
+            }
+            if (!is_null($default)) {
+                $fieldsToUpdate['default_on'] = $default;
+            }
+            if (!is_null($minimal_quantity)) {
+                $fieldsToUpdate['minimal_quantity'] = $minimal_quantity;
+            }
+            if (!is_null($available_date)) {
+                $fieldsToUpdate['available_date'] = $available_date;
+            }
+            $combination->setFieldsToUpdate($fieldsToUpdate);
```

Comment:
@LittleBigDev code review please :)

Reply:
Let me suggest you something sexier like :
```php
$fieldsToUpdate = compact(
    'price',
    'wholesale_price',
    'ecotax',
    'weight',
    'unit_price_impact',
    'default_on',
    'minimal_quantity',
    'available_date'
);
$fieldsToUpdate = array_filter(
    $fieldsToUpdate,
    create_function('$value', 'return isset($value);') // No anonymous function because we must support php 5.2
);
$combination->setFieldsToUpdate($fieldsToUpdate);
```
This code does exactly the same (building an array with all the not-null target vars) but reduces the NPATH.

Reply:
you got a point and I like it, thanks :)